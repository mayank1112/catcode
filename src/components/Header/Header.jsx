import { Fragment } from "react";
import Alert from '@material-ui/lab/Alert';
import { APP_TITLE } from "../../constants/lang";

const Header = ({ successMessage, errorMessage }) => (
    <Fragment>
        <h1>{APP_TITLE}</h1>
        {successMessage && <Alert severity="success">{successMessage}</Alert>}
        {errorMessage && <Alert severity="error">{errorMessage}</Alert>}
    </Fragment>
);

export default Header;
