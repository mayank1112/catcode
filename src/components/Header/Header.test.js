import Enzyme, { shallow } from 'enzyme';
import Header from './Header';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('<Header> component with no user messages should match snapshot', () => {
    const component = shallow(<Header />);
    expect(toJson(component)).toMatchSnapshot();
});
it('<Header> component with user messages should match snapshot', () => {
    const component = shallow(<Header errorMessage='mockError' successMessage='mockSuccess' />);
    expect(toJson(component)).toMatchSnapshot();
});
