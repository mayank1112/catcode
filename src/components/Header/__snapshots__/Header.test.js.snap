// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> component with no user messages should match snapshot 1`] = `
<Fragment>
  <h1>
    Catrancher
  </h1>
</Fragment>
`;

exports[`<Header> component with user messages should match snapshot 1`] = `
<Fragment>
  <h1>
    Catrancher
  </h1>
  <WithStyles(ForwardRef(Alert))
    severity="success"
  >
    mockSuccess
  </WithStyles(ForwardRef(Alert))>
  <WithStyles(ForwardRef(Alert))
    severity="error"
  >
    mockError
  </WithStyles(ForwardRef(Alert))>
</Fragment>
`;
